<?xml version='1.0'?>
<robot name="robot_model">

  <material name="red">
    <color rgba="0.8 0 0 1"/>
  </material>

  <material name="green">
    <color rgba="0 0.8 0 1"/>
  </material>

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>

  <material name="transparent">
    <color rgba="1 1 1 0.1"/>
  </material>
  
  <!-- Base Link -->
  <link name="base_link">
    <visual>
      <origin xyz="-0.045 0 0.068" rpy="0 0 0" />
      <geometry>
	      <box size="0.164 0.126 0.003"/>
      </geometry>
      <material name="white" />
    </visual>
    
    <!-- Base collision -->
    <collision>
      <origin xyz="-0.045 0 0.068" rpy="0 0 0" />
      <geometry>
          <box size="0.164 0.126 0.003"/>
      </geometry>
    </collision>

    <!-- Caster -->
    <visual name="caster">
      <origin xyz="-0.08 0 0.032" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.032" />
      </geometry>
      <material name="black" />
    </visual> 
    
    <!-- Caster collision -->
    <collision>
      <origin xyz="-0.08 0 00.032" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.032" />
      </geometry>
    </collision>

  </link>
  
  <!-- Right Wheel -->
  <link name="right_wheel">
    <visual>
      <origin xyz="0 0 0" rpy="1.570795 0 0" />
      <geometry>
          <cylinder length="0.01" radius="0.032" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="1.570795 0 0" />
      <geometry>
          <cylinder length="0.027" radius="0.032" />
      </geometry>
    </collision>
  </link>

  <joint name="front_right_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="right_wheel"/>
    <origin xyz="0 -0.08 0.032" rpy="0 0 0" /> 
    <axis xyz="0 1 0" />
  </joint>

  <!-- Left Wheel -->
  <link name="left_wheel">
    <visual>
      <origin xyz="0 0 0" rpy="1.570795 0 0" />
      <geometry>
          <cylinder length="0.01" radius="0.032" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="1.570795 0 0" />
      <geometry>
          <cylinder length="0.027" radius="0.032" />
      </geometry>
    </collision>
  </link>

  <joint name="front_left_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="left_wheel"/>
    <origin xyz="0 0.08 0.032" rpy="0 0 0" /> 
    <axis xyz="0 1 0" />
  </joint>

  <!-- Right Wheel Transmission -->
  <transmission name="front_right_wheel_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="front_right_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="front_right_wheel_motor">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <!-- Left Wheel Transmission -->
  <transmission name="front_left_wheel_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="front_left_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="front_left_wheel_motor">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  
</robot>